src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,122 @@
 package core.basesyntax;
 
+import java.util.ArrayList;
+import java.util.List;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+
+    private List<Entry<K, V>>[] buckets;
+    private int size;
+
+    public MyHashMap() {
+        buckets = new List[DEFAULT_CAPACITY];
+        for (int i = 0; i < DEFAULT_CAPACITY; i++) {
+            buckets[i] = new ArrayList<>();
+        }
+        size = 0;
+    }
 
     @Override
     public void put(K key, V value) {
+        int bucketIndex = getBucketIndex(key);
+        List<Entry<K, V>> bucket = buckets[bucketIndex];
+
+        if (key == null) {
+            for (Entry<K, V> entry : bucket) {
+                if (entry.getKey() == null) {
+                    entry.setValue(value);
+                    return;
+                }
+            }
+            bucket.add(new Entry<>(null, value));
+            size++;
+        } else {
+            for (Entry<K, V> entry : bucket) {
+                if (key.equals(entry.getKey())) {
+                    entry.setValue(value);
+                    return;
+                }
+            }
+            bucket.add(new Entry<>(key, value));
+            size++;
+        }
 
+        if ((double) size / buckets.length > LOAD_FACTOR) {
```

Comment:
create method for that
```suggestion
        if (needToResize()) {
```